### PR TITLE
Trainer

### PR DESCRIPTION
**Pull request by JanLin0817**
* Unify example train_cifar10.py and  train_cifar10_by_iter.py behavior
*  Distinguish train/val text logger
* Update loss logger to one line, Before was multi-lines

### CODE REVIEW

File: examples/train_cifar10.py [hunk 1]
```diff
@@ -36,18 +36,18 @@ def train_step(self, batch_data):
...
     def val_step(self, batch_data):
         inputs, labels = batch_data
 
         # forward
         outputs = self(inputs)
-        loss = criterion(outputs, labels)
+        losses = criterion(outputs, labels)
 
-        return dict(cls_loss=loss)
+        return dict(cls_loss=losses, box_loss=losses+10.0)
```
Comment (mousyball) [inline examples/train_cifar10.py:53]: Is 'box_loss' just used for demo?

File: examples/train_cifar10.py [hunk 2]
```diff
@@ -90,9 +86,9 @@ def dataloader():
     model = Net()
 
     # dataloader
-    train_loader, val_loader, classes = dataloader()
+    val_loader, classes = dataloader()
```
Comment (mousyball) [inline examples/train_cifar10.py:92]: Sync the name between testloader and val_loader at line 89 and 76.
Choose one of them and the naming style as well.

File: examples/train_cifar10.py [hunk 3]
```diff
@@ -90,9 +86,9 @@ def dataloader():
...
     train_loader = val_loader
-
+    print(len(train_loader))
```
Comment (mousyball) [inline examples/train_cifar10.py]: Is it necessary to print, or could be removed?

File: examples/train_cifar10_by_iter.py [hunk 4]
```diff
@@ -77,11 +73,13 @@ def dataloader():
     classes = ('plane', 'car', 'bird', 'cat',
                'deer', 'dog', 'frog', 'horse', 'ship', 'truck')
 
-    return trainloader, testloader, classes
+    return testloader, classes
 
 
 if __name__ == "__main__":
     # load config
+    torch.cuda.empty_cache()
```
Comment (mousyball) [inline examples/train_cifar10_by_iter.py]: Used for?

File: examples/train_cifar10.py (hunk 1 above)
Comment (JanLin0817) [reply examples/train_cifar10.py:53]: Yes, demo returning for multiple loss

File: examples/train_cifar10.py (hunk 3 above)
Comment (JanLin0817) [reply examples/train_cifar10.py]: Should be removed

File: examples/train_cifar10_by_iter.py (hunk 4 above)
Comment (JanLin0817) [reply examples/train_cifar10_by_iter.py]: Should be removed
